packages/amplify-cli/src/lib/migrate-project.js

```diff
@@ -0,0 +1,188 @@
+  const frontendPlugin = frontendPluginPath.split('/')[frontendPluginPath.split('/').length - 1];
+  const frontend = frontendPlugin.split('-')[frontendPlugin.split('-').length - 1];
+
+  newProjectConfig.frontend = frontend;
+  delete newProjectConfig.frontendHandler;
+
+  // Modify provider handler
+  const provider = Object.keys(projectConfig.providers)[0];
+  newProjectConfig.providers = [provider];
+
+  const jsonString = JSON.stringify(newProjectConfig, null, 4);
+  fs.writeFileSync(projectConfigFilePath, jsonString, 'utf8');
+}
+
+function generateLocalEnvInfo(projectConfig) {
+  const envInfo = {
+    projectPath: projectConfig.projectPath,
+    defaultEditor: projectConfig.defaultEditor,
+    envName: 'NONE',
+  };
+
+  const jsonString = JSON.stringify(envInfo, null, 4);
+  const localEnvFilePath = getLocalEnvFilePath();
+  fs.writeFileSync(localEnvFilePath, jsonString, 'utf8');
+}
+
+function generateAwsLocalInfo() {
+  const dotConfigDirPath = getDotConfigDirPath();
+  const awsInfoFilePath = path.join(dotConfigDirPath, 'aws-info.json');
+  if (fs.existsSync(awsInfoFilePath)) {
+    const awsInfo = JSON.parse(fs.readFileSync(awsInfoFilePath));
+    awsInfo.configLevel = 'project'; // Old version didn't support "General" configuation
+    const newAwsInfo = { NONE: awsInfo };
+
+    const jsonString = JSON.stringify(newAwsInfo, null, 4);
+    const localAwsInfoFilePath = path.join(dotConfigDirPath, 'local-aws-info.json');
+    fs.writeFileSync(localAwsInfoFilePath, jsonString, 'utf8');
+    fs.removeSync(awsInfoFilePath);
+  }
+}
+
+function generateTeamProviderInfo() {
+  const amplifyMetafilePath = getAmplifyMetaFilePath();
+  const amplifyMeta = JSON.parse(fs.readFileSync(amplifyMetafilePath));
+
+  const teamProviderInfo = { NONE: amplifyMeta.providers };
+
+  const jsonString = JSON.stringify(teamProviderInfo, null, 4);
+  const teamProviderFilePath = getProviderInfoFilePath();
+  fs.writeFileSync(teamProviderFilePath, jsonString, 'utf8');
+}
+
+function generateBackendConfig() {
+  const amplifyMetafilePath = getAmplifyMetaFilePath();
+  const amplifyMeta = JSON.parse(fs.readFileSync(amplifyMetafilePath));
+
+  const backendConfig = {};
+  Object.keys(amplifyMeta).forEach((category) => {
+    if (category !== 'providers') {
+      backendConfig[category] = {};
+      Object.keys(amplifyMeta[category]).forEach((resourceName) => {
+        backendConfig[category][resourceName] = {};
+        backendConfig[category][resourceName].service = amplifyMeta[category][resourceName].service;
+        backendConfig[category][resourceName].providerPlugin =
+        amplifyMeta[category][resourceName].providerPlugin;
+      });
+    }
+  });
+
+
+  const jsonString = JSON.stringify(backendConfig, null, 4);
+  const backendConfigFilePath = getBackendConfigFilePath();
+  fs.writeFileSync(backendConfigFilePath, jsonString, 'utf8');
+}
+
+
+function generateGitIgnoreFile() {
+  const getGitIgnoreAppendString = () => {
```

Comment:
Isn't there a function already which generates `.gitIgnore`

Reply:
Yeah, but it uses context.

Reply:
can we make it not to use the context

Reply:
Moving to a helper function.